torch/autograd/profiler.py

```diff
@@ -101,55 +101,49 @@ def export_chrome_trace(self, path):
+
                 for k in evt.kernels:
                     # 's' and 'f' draw Flow arrows from
                     # the CPU launch to the GPU kernel
-                    chrome_events.append(dict(
-                        name=evt.name,
-                        ph='s',
-                        ts=evt.cpu_interval.start,
-                        tid=evt.thread,
-                        pid='CPU functions',
-                        id=next_id,
-                        cat='cpu_to_cuda',
-                        args={},
-                    ))
-                    chrome_events.append(dict(
-                        name=k.name,
-                        ph='f',
-                        ts=k.interval.start,
-                        tid=k.device,
-                        pid='CUDA functions',
-                        id=next_id,
-                        cat='cpu_to_cuda',
-                        args={},
-                    ))
-                    chrome_events.append(dict(
-                        name=k.name,
-                        ph='X',
-                        ts=k.interval.start,
-                        dur=k.interval.elapsed_us(),
-                        tid=k.device,
-                        pid='CUDA functions',
-                        args={},
-                    ))
+                    f.write("{\"name\": \"" + str(evt.name) + "\", "
+                            + "\"ph\": \"s\", "
+                            + "\"ts\": " + str(evt.cpu_interval.start) + ", "
+                            + "\"tid\": " + str(evt.thread) + ", "
+                            + "\"pid\": \"CPU functions\", "
+                            + "\"id\": " + str(next_id) + ", "
+                            + "\"cat\": \"cpu_to_cuda\", "
+                            + "\"args\": {}" + "}, ")         
+                    f.write("{\"name\": \"" + str(k.name) + "\", "
+                            + "\"ph\": \"f\", "
+                            + "\"ts\": " + str(k.interval.start) + ", "
+                            + "\"tid\": " + str(k.device) + ", "
+                            + "\"pid\": \"CUDA functions\", "
+                            + "\"id\": " + str(next_id) + ", "
+                            + "\"cat\": \"cpu_to_cuda\", "
+                            + "\"args\": {}" + "}, ")
+                    f.write("{\"name\": \"" + str(k.name) + "\", "
+                            + "\"ph\": \"X\", "
+                            + "\"ts\": " + str(k.interval.start) + ", "
+                            + "\"dur\": " + str(k.interval.elapsed_us()) + ", "
+                            + "\"tid\": " + str(k.device) + ", "
+                            + "\"pid\": \"CUDA functions\", "
+                            + "\"args\": {}" + "}, ")
                     next_id += 1
-
-            json.dump(chrome_events, f)
+            f.seek(f.tell()-2, os.SEEK_SET)
+            f.write('')
+            f.write("]")
```

Comment:
It's worth writing a little comment saying explicitly what you're doing here: deleting the trailing comma, right?! Also, I'm not sure this is defined behavior: the Python docs don't mention this as a permissible operation, and POSIX doesn't seem to say what should happen in this case. Do you have authoritative docs saying this is OK? Otherwise, better to just do it the old fashioned way!

Reply:
The docs state that the seeking technique is not fine if you have a string enconding with unevenly spaced characters. However, since we're only dealing with a space and comma, I think it works fine.

Reply:
Link me?

Reply:
Can you please link to the place in the documentation where it states this?

Reply:
This SO answer says it moves the cursor byte by byte: https://stackoverflow.com/a/17023440/3992369

Reply:
And since `,` and ` ` are ASCII always single byte (and I'm controlling the output explicitly) I think this seeking technique should work fine.

Reply:
But it doesn't say that deleting a byte by passing an empty string is kosher ;)

Reply:
Also, a StackOverflow answer is hardly an authoritative source on defined behavior or not.

Reply:
Ah OK. So I could not find a good source to cite use of write so I changed to `truncate` instead.
https://docs.python.org/3/library/io.html#io.IOBase.truncate